README.md

```diff
@@ -22,7 +20,210 @@ Or install it yourself as:
+
+    post = Post.find(1234)
+    post.user_id = 5678
+    post.save!
+    exporter = YourExporter.new("fixtures/")
+    exporter.export_models([post], :user)
+    exporter.export_models([post.user])
+
+    # In our posts.yml file
+    posts_1234:
+      ...
+      user: users_5678
+
+    # In our users.yml file
+    users_5678:
+      ...
+
+When we import these fixtures using the correct `insert_order` (`['users', 'posts']`), the newly imported post will belong to the newly imported user, no matter what their new ids are.
+
+Currently, this only works for associations that return a single record (:belongs_to and :has_one).
+
+### Templates
+
+Templates allow you to export fixtures containing erb, that will be evaluated at the time of fixture import.
+
+    posts = Post.where(user_id: 5678
+    exporter = YourExporter.new("fixtures/")
+    exporter.export_models([post]) do |post|
+      { unique_promo_code: Template.new("<%= PromoCodeGenerator.unique_promo_code(#{post.id}) %>")
+    end
+
+    # In our fixtures/posts.yml file
+    posts_1234:
+      ...
+      user_id: 5678
+      unique_promo_code: <%= PromoCodeGenerator.unique_promo_code(1234) %>
+
+In the example above, we'd be able to generate a new unique promo code for each post as we import them.
+
+
+## Motivation
+
+ActiveRecord::Fixtures is designed to import data into a test database, and its practices of truncating tables and randomly generating primary keys work well in that setting.
+
+LiveFixtures is designed for importing data into a production database, where we cannot truncate the tables and where inserting records with randomly generated primary keys will cause collisions and waste space.
+
+### Here's how ActiveRecord::Fixtures work
+
+Each record is assigned a label in its yml file. Primary key values are
+assigned using a guid algorithm that maps a label to a consistent integer
+between 1 and 2^30-1. Primary keys can then be assigned before saving any
+records to the db.
+
+Why would they do this? Because, this enables us to use labels in the
+Fixture yml files to refer to associations. For example:
+
+      <users.yml>
+      bob:
+        username: thebob
+
+      <posts.yml>
+      hello:
+        message: Hello everyone!
+        user: bob
+
+
+
+The ActiveRecord::Fixture system first converts every instance of `bob` and
+`hello` into an integer using ActiveRecord::Fixture#identify, and then can
+save the records IN ANY ORDER and know that all foreign keys will be valid.
+
+There is a big problem with this. In a test db, each table is empty and so the
+odds of inserting a few dozen records causing a primary key collision is
+very small. However, for a production table with a hundred million rows, this
+is no longer the case! Collisions abound and db insertion fails.
+
+Also, auto-increment primary keys will continue from the LARGEST existing
```

Comment:
❤️ I agree! 🤐 Don't tell, but I didn't write it, I just moved it around. Maybe it was @jleven or @raorao ?